h1st/core/ensemble.py

```diff
@@ -1,107 +1,182 @@
+    This is the base class for stack ensemble classifiers
+    """
 
-    def train(self, prepared_data):
+    def evaluate(self, data: Dict, metrics: List[str]=None) -> Dict:
         """
-        Trains the StackEnsemble.
+        Evaluates for the test data
+        :param data: a dictionary {'X_test': ..., 'y_test': ...}
+        :param metrics: list of metrics to return and to persist later by the model.
+            Default value = ['confusion_matrix', 'precision', 'recall', 'f1', 'support', 'accuracy']
+        
+        :return:
+            a dictionary containing requested metrics
 
-        :param prepared_data: Dictionary object contains train data, labels and validation data.
         """
-        train_data = prepared_data['train_data']
-        train_labels = prepared_data['train_labels']
-        all_inputs = None
-        all_labels = None
+        def add_metric(name, value):
+            if name in metrics:
+                self.metrics[name] = value
+
+        if not metrics:
+            metrics = ['confusion_matrix', 'precision', 'recall', 'f1', 'support', 'accuracy']
+
+        X_test, y_test = data['X_test'], data['y_test']
+        y_pred = self.predict({'X': X_test})['predictions']     
         
-        for each_input, each_label in zip(train_data, train_labels):
-            inputs = None
-            labels = each_label
-            for model in self.models:
-                df_prediction = self._extract_prediction(model.predict(each_input))
-                
-                df_prediction = df_prediction[self.prediction_columns]
-                df_prediction.fillna(0, inplace=True)
-                p1 = df_prediction.values
-                if inputs is None:
-                    inputs = p1
-                else:
-                    inputs = np.hstack((inputs, p1))
-            if all_inputs is None:
-                all_inputs = inputs
-            else:
-                all_inputs = np.vstack((all_inputs, inputs))
-            if all_labels is None:
-                all_labels = labels
-            else:
-                all_labels = np.vstack((all_labels, labels))
-
-        # assumption: all_inputs ~ numpy array, all_labels ~ numpy array
-        self.model.fit(all_inputs, all_labels)
-
-
-class MultiOutputClassifierEnsemble(StackEnsemble):
-    def __init__(self, models):
-        ensembler = MultiOutputClassifier(RandomForestClassifier(n_jobs=-1, max_depth=4, random_state=42))
-        super().__init__(models, ensembler)
+        precision, recall, f1, support = precision_recall_fscore_support(y_test, y_pred)
+
+        add_metric('confusion_matrix', confusion_matrix(y_test, y_pred))        
+        add_metric('precision', precision)
+        add_metric('recall', recall)
+        add_metric('f1', f1)
+        add_metric('support', support)
+        add_metric('accuracy', accuracy_score(y_test, y_pred))
+
+        return self.metrics
+
+
+class RandomForestStackEnsembleClassifier(StackEnsembleClassifier):
```

Comment:
The class hierarchy here is confusing (and incorrect). It should be, e.g., 

`Ensemble` → `StackEnsemble` → `ClassifierStackEnsemble` → `RandomForestClassifierStackEnsemble`

Also please use one file per class.